python/tests/test_table_transforms.py

```diff
@@ -0,0 +1,420 @@
+# MIT License
+#
+# Copyright (c) 2022 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Test cases for table transformation operations like trim(), decapitate, etc.
+"""
+import numpy as np
+import pytest
+
+import tests
+import tskit
+from tests.test_highlevel import get_example_tree_sequences
+
+# ↑ See https://github.com/tskit-dev/tskit/issues/1804 for when
+# we can remove this.
+
+
+def split_edges_definition(ts, time, *, flags=None, population=None, metadata=None):
+    tables = ts.dump_tables()
+    if ts.num_migrations > 0:
+        raise ValueError("Migrations not supported")
+    default_population = population is None
+    if not default_population:
+        # -1 is a valid value
+        if population < -1 or population >= ts.num_populations:
+            raise ValueError("Population out of bounds")
+    flags = 0 if flags is None else flags
+    if metadata is None:
+        metadata = tables.nodes.metadata_schema.empty_value
+    metadata = tables.nodes.metadata_schema.validate_and_encode_row(metadata)
+    # This is the easiest way to turn off encoding when calling add_row below
```

Comment:
Does this pattern here imply we should have a way to pass in encoded metadata to `add_row`? Maybe `encoded_metadata=`?

Reply:
I think so - either that or have a boolean flag for controlling whether we encode or not?